samples/apps/copilot-chat-app/webapp/README.md

```diff
@@ -5,6 +5,73 @@ production deployments.
 
 See [../README.md](../README.md) for complete instructions on setting up and running the application.
 
+## How to use HTTPS for local development
+If you want to run Copilot Chat with HTTPS, you need to create a certificate and sign it with a Certificate Authority (CA) that is trusted locally by your device and browser. You can do that easily with [mkcert](https://github.com/FiloSottile/mkcert#installation).
+
+1. From an elevated shell, run
+
+    ### Windows
+    On Windows, use [Chocolatey](https://chocolatey.org/)
+    ```
+    choco install mkcert
+    ```
+
+    ### MacOS
+    On macOS, use [Homebrew](https://brew.sh/)
+    ```
+    brew install mkcert
+    ```
+
+    For installation on other OS and advanced topics (i.e., supported root stores), see the [official mkcert installation guide](https://github.com/FiloSottile/mkcert#installation). 
+
+1. Create a new local CA by running
+    ```
+    mkcert -install
+    ```
+1. Create a new certificate with all hostnames you wish to run the app on. 
+    > It is recommend you do this at the /WebApp directory level. 
+
+    Run
+    ```
+    mkcert localhost 127.0.0.1 example.test
+    ```
+    You should see an output that looks like this:
+    ```
+    Created a new certificate valid for the following names 📜
+        - "localhost"
+        - "127.0.0.1"
+        - "example.test"
+
+    The certificate is at "./localhost+1.pem" and the key at "./localhost+1-key.pem" ✅
+
+    It will expire on 1 September 2025
+    ```
+
+    > **Warning**: (from the developers of mkcert) the rootCA-key.pem file that mkcert automatically generates gives complete power to intercept secure requests from your machine. Do not share it.
+1. In the WebApp `.env` file, uncomment the following lines and populate with your respective certificate and key files generated in the step above.
+    ```
+    ...
+    # To enable HTTPS, uncomment the following lines
+    HTTPS="true"
+    SSL_CRT_FILE=localhost+1.pem # Replace with your locally-trusted cert file
+    SSL_KEY_FILE=localhost+1-key.pem # Replace with your locally-trusted cert key
+    ```
+1. In the [webapi appsettings.json](../webapi/appsettings.json) file, find the `"AllowedOrigins"` section, and add the URLs (with ports!) that you'll be running the apps on already with `https` prefixed. i.e., from the example hostnames above, the `"AllowedOrigins"` section should look like:
+    ```
+    ...
+    // CORS
+    "AllowedOrigins": [
+        "http://localhost:3000", 
+        "https://localhost:3000",
+        "https://127.0.0.1:3000",
+        "https://example.test:3000"
+    ],
+    ...
+    ```
+1. Add the same URLs (with ports!) as SPA redirect URIs to your AAD app registration. This can be done in Azure Portal ([https://portal.azure.com](https://ms.portal.azure.com/#home)).
```

Comment:
```suggestion
1. Add the same URLs (with ports!) as single-page authority (SPA) redirect URIs to your Azure Active Directory (AAD) application registration. This can be done in the [Azure Portal](https://portal.azure.com).
```

Reply:
SPA stands for Single Page Application, not authority!